Iroh console does not detect when the remote node goes away and comes back

You can leave an iroh console session open while restarting the node. This is great. Very useful.

However, the first two commands after the node has been restarted will fail. See interaction below.

![image](https://github.com/n0-computer/iroh/assets/248257/e68ed352-9e08-47fc-a401-ce22d6799fc6)

It would be nice if we could avoid this, e.g. by having some sort of keepalive on the connection that detects quickly that the other side went away. This might require a small config change in quic-rpc.

i tried add tracing fmt-source-locations to iroh-main and it ident quick-rpc as writing the err probably before iroh has a chance catch it ?

`
/quic-rpc-0.6.1/src/transport/quinn.rs:341 
error opening bidi substream: closed by peer: server endpoint dropped (code 0)
`

```
async fn main_impl() -> Result<()> {
    let lp = tokio_util::task::LocalPoolHandle::new(num_cpus::get());
    // tracing_subscriber::registry()
    //     .with(tracing_subscriber::fmt::layer().with_writer(std::io::stderr))
    //     .with(EnvFilter::from_default_env())
    //     .init();

    tracing_subscriber::fmt()
        // enable everything
        .with_max_level(tracing::Level::DEBUG)
        // display source code file paths
        .with_file(true)
        // display source code line numbers
        .with_line_number(true)
        // disable targets
        .with_target(false)
        // sets this to be the default, global collector for this application.
        .init();

    let cli = Cli::parse();
    cli.run(lp).await
}
```



```
blob list blobs2024-01-07T13:20:51.679666Z DEBUG rustyline: /Users/xraid/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustyline-12.0.0/src/tty/unix.rs:791: c: '\r' => key: KeyEvent(Enter, Modifiers(0x0))    
2024-01-07T13:20:51.679820Z DEBUG rustyline: /Users/xraid/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustyline-12.0.0/src/keymap.rs:655: Emacs command: AcceptOrInsertLine { accept_in_the_middle: true }    

2024-01-07T13:20:51.681621Z DEBUG quic_rpc::transport::quinn: /Users/xraid/.cargo/registry/src/index.crates.io-6f17d22bba15001f/quic-rpc-0.6.1/src/transport/quinn.rs:332: Got request for new bidi substream
2024-01-07T13:20:51.681691Z  WARN quic_rpc::transport::quinn: /Users/xraid/.cargo/registry/src/index.crates.io-6f17d22bba15001f/quic-rpc-0.6.1/src/transport/quinn.rs:341: error opening bidi substream: closed by peer: server endpoint dropped (code 0)
2024-01-07T13:20:51.681715Z  WARN quic_rpc::transport::quinn: /Users/xraid/.cargo/registry/src/index.crates.io-6f17d22bba15001f/quic-rpc-0.6.1/src/transport/quinn.rs:342: recreating connection
Error: Open(LocallyClosed)
2024-01-07T13:20:51.681880Z DEBUG quic_rpc::transport::quinn: /Users/xraid/.cargo/registry/src/index.crates.io-6f17d22bba15001f/quic-rpc-0.6.1/src/transport/quinn.rs:305: Connecting to 127.0.0.1:4919 as localhost
```

--- a/iroh/Cargo.toml
+++ b/iroh/Cargo.toml
@@ -39,7 +39,7 @@ iroh-gossip = { version = "0.12.0", path = "../iroh-gossip" }
 once_cell = "1.18.0"
 parking_lot = "0.12.1"
 postcard = { version = "1", default-features = false, features = ["alloc", "use-std", "experimental-derive"] }
-quic-rpc = { version = "0.6", default-features = false, features = ["flume-transport"] }
+quic-rpc = { version = "0.6.2", default-features = false, features = ["flume-transport"] }
 quinn = "0.10"
 range-collections = { version = "0.4.0" }
 rand = "0.8"

